src/fu-engine.c

```diff
@@ -3889,6 +3889,18 @@ fu_engine_sort_devices_by_priority_name (gconstpointer a, gconstpointer b)
 	const gchar *name_a = fu_device_get_name (dev_a);
 	const gchar *name_b = fu_device_get_name (dev_b);
 
+	/* first sort by flags (so clients can display groupings ) */
+	if ((!fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_UPDATABLE) &&
+	    fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_UPDATABLE)) ||
+	    (!fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_SUPPORTED) &&
+	    fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_SUPPORTED)))
+		return -1;
+	if ((fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_UPDATABLE) &&
+	    !fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_UPDATABLE)) ||
+	    (fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_SUPPORTED) &&
+	    !fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_SUPPORTED)))
+		return 1;
+	/* then support by priority, so clients can show parents */
```

Comment:
i'm not super keen on this being in the daemon; at the moment we can say the devices are ordered "by priority then name" but with this change it all gets a bit more fuzzy.

Reply:
Adjusted